### PR TITLE
tests : add periph_gpio_arduino for verifying arduino pin mappings

### PR DESCRIPTION
**Pull request by smlng**
### Contribution description

copy with squashed version of #9707, retaining original authorship by @TomKeddie

### Testing procedure

compile and flash `tests/periph_gpio_arduino` onto your favourite Arduino board and e.g. toggle the LED with

```
init_out 13
toggle 13
toggle 13
```

### Issues/PRs references

see also #9707

### CODE REVIEW

**Comment by smlng** [comment]
advise original author on how to squash is better I think

**Comment by smlng** [comment]
replaces #9707, squashed and rebases while retaining authorship of @TomKeddie

File: tests/periph_gpio_arduino/Makefile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+include ../Makefile.tests_common
+
+FEATURES_REQUIRED = periph_gpio
```
Comment (maribu) [inline tests/periph_gpio_arduino/Makefile]: Hmm, I would have guessed that `FEATURES_REQUIRED += arduino` would be required. But compilation now only fails for the ATmega328p based boards. (Due to lack of ROM/RAM.)

**Comment by maribu** [comment]
Tested on Arduino Mega2560 and Arduino Due

**Comment by PeterKietzmann** [comment]
@smlng please address murdock errors

`tests/periph_gpio_arduino/README.md:68: new blank line at EOF.`

File: tests/periph_gpio_arduino/Makefile (hunk 1 above)
Comment (smlng) [reply tests/periph_gpio_arduino/Makefile]: the `USEMODULE += arduino` will pull in the `arduino` feature, see `Makefile.dep` - that's how it is handled for other modules too.

However, to me this is not consistent or logically sound in all places, so you got a point there. But also: that's another PR to, if we want to change something there.

**Comment by smlng** [comment]
@PeterKietzmann CI errors are fixed:

- amended fix to README directly
- added boards with insufficient memory as an extra commit as suggested

File: tests/periph_gpio_arduino/Makefile (hunk 1 above)
Comment (maribu) [reply tests/periph_gpio_arduino/Makefile]: > the `USEMODULE += arduino` will pull in the `arduino` feature

Aha! Thanks for the explanation :-)

**Comment by smlng** [comment]
also thanks to @TomKeddie for the original work

**Pull request merged**